opensearch/catalog-opensearch-endpoint/src/main/java/org/codice/ddf/endpoints/ResourceUriTransformerCatalogFrameworkProxy.java

```diff
@@ -0,0 +1,199 @@
+        ResourceNotFoundException, ResourceNotSupportedException {
+        return catalog.getEnterpriseResource(request);
+    }
+
+    public Map<String, Set<String>> getEnterpriseResourceOptions(String metacardId)
+        throws ResourceNotFoundException {
+        return catalog.getEnterpriseResourceOptions(metacardId);
+    }
+
+    public ResourceResponse getLocalResource(ResourceRequest request) throws IOException,
+        ResourceNotFoundException, ResourceNotSupportedException {
+        return catalog.getLocalResource(request);
+    }
+
+    public Map<String, Set<String>> getLocalResourceOptions(String metacardId)
+        throws ResourceNotFoundException {
+        return catalog.getLocalResourceOptions(metacardId);
+    }
+
+    public ResourceResponse getResource(ResourceRequest request, String resourceSiteName)
+        throws IOException, ResourceNotFoundException, ResourceNotSupportedException {
+        return catalog.getResource(request, resourceSiteName);
+    }
+
+    public Map<String, Set<String>> getResourceOptions(String metacardId, String sourceId)
+        throws ResourceNotFoundException {
+        return catalog.getResourceOptions(metacardId, sourceId);
+    }
+
+    public Set<String> getSourceIds() {
+        return catalog.getSourceIds();
+    }
+
+    public SourceInfoResponse getSourceInfo(SourceInfoRequest sourceInfoRequest)
+        throws SourceUnavailableException {
+        return catalog.getSourceInfo(sourceInfoRequest);
+    }
+
+    public QueryResponse query(QueryRequest query) throws UnsupportedQueryException,
+        SourceUnavailableException, FederationException {
+        return catalog.query(query);
+    }
+
+    public QueryResponse query(QueryRequest queryRequest, FederationStrategy strategy)
+        throws SourceUnavailableException, UnsupportedQueryException, FederationException {
+        return catalog.query(queryRequest, strategy);
+    }
+
+    public BinaryContent transform(Metacard metacard, String transformerId,
+            Map<String, Serializable> requestProperties) throws CatalogTransformerException {
+        return catalog.transform(metacard, transformerId, requestProperties);
+    }
+
+    /**
+     * Converts the resource URIs found in the {@link SourceResponse} {@link Metacard}s using the
+     * {@link ActionProvider} injected in the constructor before delegating the call to the proxied
+     * {@link CatalogFramework} object.
+     */
+    public BinaryContent transform(SourceResponse response, String transformerId,
+            Map<String, Serializable> requestProperties) throws CatalogTransformerException {
+        transform(response);
+        return catalog.transform(response, transformerId, requestProperties);
+    }
+
+    public UpdateResponse update(UpdateRequest updateRequest) throws IngestException,
+        SourceUnavailableException {
+        return catalog.update(updateRequest);
+    }
+
+    private void transform(SourceResponse queryResponse) {
```

Comment:
That's a good point @pklinef about clients not always being endpoints.

During the brainstorm process did you guys look at adding the external url as a different attribute in the metacard (instead of overwriting the current uri)? That would leverage the extensibility of the metacard interface, not add code dependencies in the endpoints (i.e. require using the proxy for future endpoints), and maintain the internal uri that may be of use in future components. It would be a very minor change to the current transformers and allows the possibility of using the internal uri as a fallback for backwards compatibility.